File: server/src/main/java/bit/project/server/entity/Salary.java
package bit.project.server.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Salary {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDateTime tocreation;

    @Lob
    private String description;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;

    @ManyToOne
    private Employee employee;

    private LocalDate date;

    private BigDecimal amount;

    public Salary(Integer id, String code) {
        this.id = id;
        this.code = code;
    }

    public Salary(Integer id, String code, Employee employee, LocalDate date, BigDecimal amount) {
        this.id = id;
        this.code = code;
        this.employee = employee;
        this.date = date;
        this.amount = amount;
    }

}

File: client/src/app/entities/salary.ts

import {User} from './user';
import {DataPage} from '../shared/data-page';
import {Category} from './category';
import {Itemstatus} from './itemstatus';
import {Employee} from './employee';

export class Salary{
  id: number;
  code: string;
  tocreation: string;
  description: string;
  creator: User;
  employee: Employee;
  date: string;
  amount: number;

  constructor(id: number = null) {
    this.id = id;
  }
}

export class SalaryDataPage extends DataPage{
  content: Salary[];
}
